
<h2>Problem</h2>
<p>Detailed function trace information does not appear when using <code>asyncio.ensure_future</code> in coroutines.</p>
<h2>Solution</h2>
<p>Futures created from <code>ensure_future</code> must be awaited in the same coroutine in which they've been created. For example,in the <strong>Before</strong> section, <code>await</code> is not present with <code>ensure_future</code>, which would result in missing information:</p>
<p><strong>Before</strong>:</p>
<pre><code>import asyncio

async def foo():
  ensure_future(bar())

async def bar():
  await asyncio.sleep(0.5)
</code></pre>
<p><strong>After</strong>:</p>
<pre><code>import asyncio

async def foo():
  await ensure_future(bar())

async def bar():
  await asyncio.sleep(0.5)
</code></pre>
